src/extension/linkify/common/modelFilePathLinkifier.ts

```diff
@@ -0,0 +1,239 @@
+	}
+
+	private normalizeSlashes(value: string): string {
+		// Collapse one or more backslashes into a single forward slash so mixed separators normalize consistently.
+		return value.replace(/\\+/g, '/');
+	}
+
+	private normalizeLinkText(rawText: string): string {
+		let text = this.normalizeSlashes(rawText);
+		// Remove a leading or trailing backtick that sometimes wraps the visible link label.
+		text = text.replace(/^`|`$/g, '');
+
+		// Look for a trailing #L anchor segment so it can be stripped before we compare names.
+		const anchorMatch = /^(.+?)(#L\d+(?:-\d+)?)$/.exec(text);
+		return anchorMatch ? anchorMatch[1] : text;
+	}
+
+	private canLinkify(parsed: { readonly text: string; readonly targetPath: string; readonly anchor: string | undefined }, workspaceFolders: readonly Uri[]): boolean {
+		const { text, targetPath, anchor } = parsed;
+		const textMatchesBase = targetPath === text;
+		const textIsFilename = !text.includes('/') && targetPath.endsWith(`/${text}`);
+		const descriptiveWithAnchor = !!anchor; // Allow any descriptive text when anchor is present
+
+		return Boolean(workspaceFolders.length) && (textMatchesBase || textIsFilename || descriptiveWithAnchor);
+	}
+
+	private async resolveTarget(targetPath: string, workspaceFolders: readonly Uri[], preserveDirectorySlash: boolean): Promise<Uri | undefined> {
+		if (!workspaceFolders.length) {
+			return undefined;
+		}
+
+		const folderUris = workspaceFolders.map(folder => this.toVsUri(folder));
+
+		if (this.isAbsolutePath(targetPath)) {
+			const absoluteUri = this.tryCreateFileUri(targetPath);
+			if (!absoluteUri) {
+				return undefined;
+			}
+
+			for (const folderUri of folderUris) {
+				if (this.isEqualOrParentFs(absoluteUri, folderUri)) {
+					return this.tryStat(absoluteUri, preserveDirectorySlash);
+				}
+			}
+			return undefined;
+		}
+
+		const segments = targetPath.split('/').filter(Boolean);
+		for (const folderUri of folderUris) {
+			const candidate = Uri.joinPath(folderUri, ...segments);
+			const stat = await this.tryStat(candidate, preserveDirectorySlash);
+			if (stat) {
+				return stat;
+			}
+		}
+
+		return undefined;
+	}
+
+	private tryCreateFileUri(path: string): Uri | undefined {
+		try {
+			return Uri.file(path);
+		} catch {
+			return undefined;
+		}
+	}
+
+	private toVsUri(folder: Uri): Uri {
+		return Uri.parse(folder.toString());
+	}
+
+	private isEqualOrParentFs(target: Uri, folder: Uri): boolean {
+		const targetFs = this.normalizeFsPath(target);
+		const folderFs = this.normalizeFsPath(folder);
+		return targetFs === folderFs || targetFs.startsWith(folderFs.endsWith('/') ? folderFs : `${folderFs}/`);
+	}
+
+	private normalizeFsPath(resource: Uri): string {
+		// Convert Windows backslashes to forward slashes and remove duplicate separators for stable comparisons.
+		return resource.fsPath.replace(/\\/g, '/').replace(/\/+/g, '/').toLowerCase();
```

Comment:
Doing this manually also seems fishy. If you want paths with just `/`, you a use `.path` instead of `.fsPath`. Also you can use `normalizePath` to normalize uris

Reply:
fixed